### DISCUSSION TITLE
What is the unit of Node.total_inflow?

### DISCUSSION THREAD

**Discussion by Aitianxia333** [pyswmm/pyswmm#548] [Q&A] [answered]
Thank you to the developers for their outstanding contributions. I would like to know if the Node.total_inflow in pyswmm calculates the total inflow at a node. If it does, what is the unit? I used the following code:

from pyswmm import Simulation, Nodes
with Simulation('swmm.inp') as sim:
    j1 = Nodes(sim)["J001"]
    for step in sim:
        print(j1.total_inflow)
The j1.total_inflow values I calculated differ from the total inflow values for J001 in the .rpt file.So I would like to know what the unit of Node.total_inflow is.Thanks

**Comment by bemcdonnell** [comment]
This value is the total inflow rate to a node.  Here is a link to the documentation.  This differs from lateral inflow.  total inflow includes both flows from the connecting links and the lateral inflow coming into that particular node (which includes DWF, RDII, runoff, etc)

https://pyswmm.github.io/pyswmm/reference/api/api/pyswmm.nodes.Node.total_inflow.html#pyswmm.nodes.Node.total_inflow

units are `length^3/t`  (whichever unit you are working in - mgd/cfs/cms/lps)

**Comment by Aitianxia333** [comment]
Hello, if I want to obtain the node inflow volume in units of 10*6 liters, similar to the results in a SWMM report, as shown in the figure below, how should I do it? Thank you for your help.
![Question](https://github.com/user-attachments/assets/1b642634-cb3c-4628-95c5-c7569436da49)

**Reply by bemcdonnell** [reply, marked as answer]
You're looking for node summary statistics: 

https://pyswmm.github.io/pyswmm/reference/api/api/pyswmm.nodes.Node.statistics.html

```python
with Simulation("inp_path.inp") as sim:
    for step in sim:
        pass
    
    node_info = Nodes(sim)["J001"]
    stats = node_info.statistics
    print(stats)
```

That unit might be ft^3.  I can't remember exactly 

Have a look at the "Latte" example on the website:  https://www.pyswmm.org/examples
